core/GlobalState.h

```diff
@@ -235,6 +235,8 @@ class GlobalState final {
     std::unique_ptr<GlobalState> deepCopy(bool keepId = false) const;
     mutable std::shared_ptr<ErrorQueue> errorQueue;
 
+    std::unique_ptr<GlobalState> copyForIndex() const;
```

Comment:
doc comment to clarify when this specifically should be used over deepCopy?

Reply:
seconding jez's comment here